smoker/server/plugins/__init__.py

```diff
@@ -458,7 +459,15 @@ def run(self):
         self.drop_privileged()
 
         with semaphore:
-            self.run_plugin(self.forced)
+            if (self.params['MaintenanceLock'] and
+                    os.path.exists(self.params['MaintenanceLock'])):
+                result = Result()
+                result.set_forced(self.forced)
+                result.set_status('OK')
```

Comment:
Skipped plugin run should be warning imho.
Otherwise you may accidentaly leave maintenance lock file and see that everything is ok in monitoring.